### ISSUE TITLE
cards, dashboards, and non-text contrast

### ISSUE THREAD

**Issue by gundulaniemann** [w3c/wcag#1945]
In dashboards and similarly arranged pages, the cards on one hand can be very complex and come in different sizes, on the other they usually are active as a whole (being focusable, clickable, having a context menu, ...).
In contrast to very regular pages, where the content is very consistent, it is kind of impossible to tell the cards apart (that is, which UI element belongs to which card) if the card does not somehow contrast to the background.

Do you agree in such cases 1.4.11 needs to be fulfilled?

Diving through existing discussions I feel this is the case. 

In [Issue #1187](https://github.com/w3c/wcag/issues/1187)
patrickhlauke said: 

> border around a card...as discussed previously (e.g. #856) this is also not really absolutely necessary in this specific example - assuming there's nothing around the card/following the card which, if the border wasn't there, could be mistaken (incorrectly) to be still part of that card. basically, it depends (tm) on the specific situation. auditors need to use their judgement if, in the specific context, something is necessary or not. there's no absolute rules about "every border must have 3:1" or similar. it really comes down to a judgement call on the part of the auditor, i'd say.

In [Issue #856](https://github.com/w3c/wcag/issues/856)
mraccess77 said

> The card border could indicate a grouping which is alone communicates with low contrast. Low vision user might miss that information.

and also

>  In the example shown I am able to tell the associations -- but I've seen other examples where without the border it wasn't clear if the text applied to the image above or below it. That is the border of the card separated regions of content that otherwise were ambiguous. I've also seen borders of content that is overlayed over top of other page contents like chat widgets. Without the border it can be unclear of the relationship of the content to a separate chat window versus content of the page in general. So perhaps other factors may need to be taken into account such as these.

I collected some example dashboards. In several of them, very similar, if not even equal, elemenst are in separated cards on one hand and in same cards on the other.
[analytic dashboard](https://www.memsource.com/blog/new-homepage-analytics-dashboards-to-track-localization/)
[data and Analytics](https://dribbble.com/shots/14630087--Data-and-Analytics)
[various dashboard templates](https://colorlib.com/wp/free-dashboard-templates/)

**Comment by JAWS-test** [comment]
While it would be nice for such cards to have a 3:1 contrast border, I don't think WCAG requires this in every case. As @patrickhlauke wrote, it depends on the context. Context would be e.g. also the distance (white space) between the cards: If this is large enough and suitable to clearly identify the cards as such (i.e. the same distance does not occur within the cards), then I think a border with 3:1 contrast is not necessary

**Comment by patrickhlauke** [comment]
and ultimately, it's about finding a balance. WCAG is, for better or worse, a lowest common denominator. having SCs that are far too overreaching (e.g. requiring every card etc to have a 3:1 contrasting border, or background that contrasts against its parent container/page background) would make that majority of web content out there today illegal overnight. i can guarantee this would simply result in formal objections, and in general make designers/developers even less likely to adopt what appears to be a far too draconian/broad brush approach.

**Comment by gundulaniemann** [comment]
I see completely it is not reasonable to request it at any rate and in any situation.
In the dashboards I have in mind (and which I tried to find as examples), the structure is not detectable by spacing only, neither by repetition of the layout.
If it is the only way to detect what belongs to which card, is a contrast required then?

**Comment by JAWS-test** [comment]
>  If it is the only way to detect what belongs to which card, is a contrast required then?

Yes, I think so

**Comment by jake-abma** [comment]
1.4.11 says:

> Graphical Objects
> Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.

So IF a graphic is used and needed to understand the content (in this case it specifically belongs to the card) it's a big YES, the problem is, I think, when a CSS border or alike is used and not seen as "a graphic"... then we have a 'gap' in the spec

**Comment by jake-abma** [comment]
[[correction: Graphical Objects]]

**Comment by detlevhfischer** [comment]
Most cards I know group their contents in the same way, so there is also a gestalt aspect that visually ties the card visually together independent of spacing or border contrast. I think this is difficult to discuss in the absence of concrete examples.

**Comment by gundulaniemann** [comment]
That's why I linked some examples.
Can the links be opened?

**Comment by detlevhfischer** [comment]
Sorry I overloooked the examples. Apart from specific issues (likely contrast of headings too low in some cases etc.) and other isues that would be identified regardless of whether these are cards or just separate sections on a page, I cannot see a definitive requrement to indicate the border of the card by default. I guess when focused as a whole, the focus indication will have to meet 3:1 against the adjacent background (with additional requirements to come with 2.4.11) - But as it stands, whether content will be perceived to be part o a card or bleed into other cards, be misinterpreted because of that etc., seems something not specifically covered by 1.4.11, in my book - more a general usability problem.

**Comment by gundulaniemann** [comment]
If it i a general usability problm to be able to tell the cards apart from each other, it becomes an Accessibility problem if it affects only specific users like users with color or contrast vision impairments.
Might it be a violation of 1.4.1 Use Color rather than a violation of 1.4.11?
The issue arises if the color or a very light border or a very slight distinction of card color and page background is the only indication for the cards.
As soon as there is significant spacing, repetitive arrangement inside the cards, cards in strictly the same sizes, to name a few possibilities, the cards can be told apart. But if none of these is true, an additional indication (like contrast) is needed.
Do you agree?

**Comment by mraccess77** [comment]
If the border or background color is necessary to understand grouping because of formatting then yes I'd say that's an issue - otherwise no.

**Comment by alastc** [comment]
Taking the first example from [various dashboard examples](https://colorlib.com/wp/free-dashboard-templates/) and removing the card borders:

<img src="https://user-images.githubusercontent.com/216630/124646785-fe68c980-de8c-11eb-96c6-63cab4781174.png" alt="Analytics dashboard example, read text below for context.">

I don't think that makes a lot of impact. You could do that exercise for most or all of the examples and get a similar result.

Also, there is a complication which is that the card is rarely a "user interface component" itself, it is usually a content container which may include user-interface components. It is not an area suited to hard rules.

**Comment by gundulaniemann** [comment]
In dashboards indeed cards often are interactive as one entity. Activating them yields to an application with details, and usually they also carry a context menu to manipulate the dashboard itself (reorder, add new cards, delete cards, change their size, ..).

**Comment by alastc** [comment]
Often, but not always. In things like Google Analytics the card itself is never interactive. (i.e. clicking in white-space within the card does nothing). Therefore there is no requirement for it to have a border (at least under the 'user-interface components' part).

![image](https://user-images.githubusercontent.com/216630/124961490-77932880-e015-11eb-903e-3e09311e86cb.png)

There are things within it that are interactive, most of which have little arrow indicators. (Although not the tabs at the top.) But the card itself is not interactive.

**Comment by fstrr** [comment]
This issue looks like it can be closed. If it needs to be re-opened, please do that and convert it to a Discussion.
